samtranslator/feature_toggle/feature_toggle.py

```diff
@@ -18,51 +19,76 @@ class FeatureToggle:
             LOG.warning("Feature '{}' not available in Feature Toggle Config.".format(feature_name))
             return False
+
+        stage = self.stage
+        region = self.region
+        account_id = self.account_id
+        if not stage or not region or not account_id:
+            LOG.warning(
+                "One or more of stage, region and account_id is not set. Feature '{}' not enabled.".format(feature_name)
+            )
+            return False
+
         stage_config = self.feature_config.get(feature_name, {}).get(stage, {})
         if not stage_config:
             LOG.info("Stage '{}' not enabled for Feature '{}'.".format(stage, feature_name))
             return False
-        region_config = stage_config.get(region, {}) if region in stage_config else stage_config.get("default", {})
-        is_enabled = region_config.get("enabled", False)
+
+        if account_id in stage_config:
+            account_config = stage_config[account_id]
+            region_config = account_config[region] if region in account_config else account_config.get("default", {})
+        else:
+            region_config = stage_config[region] if region in stage_config else stage_config.get("default", {})
+
+        is_enabled = self._is_feature_enabled_for_region_config(feature_name, region_config)
+
         LOG.info("Feature '{}' is enabled: '{}'".format(feature_name, is_enabled))
         return is_enabled
 
-    def is_enabled_for_account_in_region(self, feature_name, stage, account_id, region="default"):
+    def _get_account_percentile(self, feature_name):
         """
-        To check if feature is available for a particular account or not.
+        Get account percentile based on sha256 hash of account ID and feature_name
+
         :param feature_name: name of feature
-        :param stage: stage where SAM is running
-        :param account_id: account_id who is executing SAM template
-        :param region: region in which SAM is running
-        :return:
+        :returns: integer n, where 0 <= n < 100
         """
-        if feature_name not in self.feature_config:
-            LOG.warning("Feature '{}' not available in Feature Toggle Config.".format(feature_name))
-            return False
-        stage_config = self.feature_config.get(feature_name, {}).get(stage, {})
-        if not stage_config:
-            LOG.info("Stage '{}' not enabled for Feature '{}'.".format(stage, feature_name))
-            return False
-        account_config = stage_config.get(account_id) if account_id in stage_config else stage_config.get("default", {})
-        region_config = (
-            account_config.get(region, {}) if region in account_config else account_config.get("default", {})
-        )
-        is_enabled = region_config.get("enabled", False)
-        LOG.info("Feature '{}' is enabled: '{}'".format(feature_name, is_enabled))
+        m = hashlib.sha256()
+        m.update(self.account_id.encode())
```

Comment:
this may have been pre-validated, so its more of a question. will we always have an account_id and feature_name defined by the time we get here?

Reply:
yes, both of them are defined when we get here. `account_id` is required when instantiating the `FeatureToggle` class, and `feature_name` is a required parameter for this method.